FirebaseCombineSwift/Sources/Auth/Auth+Combine.swift

```diff
@@ -0,0 +1,485 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if canImport(Combine) && swift(>=5.0) && canImport(FirebaseAuth)
+
+  import Combine
+  import FirebaseAuth
+
+  @available(swift 5.0)
+  @available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, *)
+  extension Auth {
+    // MARK: - Authentication State Management
+
+    /// Registers a publisher that publishes authentication state changes.
+    ///
+    /// The publisher emits values when:
+    ///
+    /// - It is registered,
+    /// - A user with a different UID from the current user has signed in, or
+    /// - The current user has signed out.
+    ///
+    /// The publisher will emit events on the **main** thread.
+    ///
+    /// - Returns: A publisher emitting (`Auth`, `User`) tuples.
+    public func authStateDidChangePublisher() -> AnyPublisher<(Auth, User?), Never> {
```

Comment:
I mostly did this to match the existing API. In case one does indeed want to access the `auth` instance, it might be more straight-forward to write `auth.someMethod()` instead of `Auth.auth().someMethod()`.

Reply:
Any objections to removing the `Auth` instance from the closure, @morganchen12 / @rosalyntan ?

Reply:
I have a slight preference for matching the existing API, but no strong opinion.